### PR TITLE
Added documentation link, source code link and logo for all the default applications

### LINKED ISSUE #12942: Add documentation link, source code link, and logo for all our default applications
Dependent on:

- [x] https://github.com/kubermatic/kubermatic/issues/12937
- [x] https://github.com/kubermatic/kubermatic/issues/12938

### PR DESCRIPTION
**Pull request by archups**
**What this PR does / why we need it**:
It will add documentation link, source code link and logos to all the applications which are part of the default applications catalog.

**Which issue(s) this PR fixes**:

Fixes #12942 

**What type of PR is this?**
/kind feature

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Add documentation link, source code link and logo to the default applications.
```

**Documentation**:

```documentation
https://github.com/kubermatic/docs/pull/1631
```

### CODE REVIEW

**Comment by archups** [comment]
Dependent on https://github.com/kubermatic/dashboard/issues/6444 for verification.

**Comment by archups** [comment]
> /

@embik yes I think the script fix  is the needed! :D I will do it. Stumbled into this issue twice. Thanks! :)

**Review by embik — APPROVED** [review]
LGTM, leaving approval to @SimonTheLeg or @csengerszabo.

File: pkg/ee/default-application-catalog/applicationdefinitions/argocd-app.yaml [hunk 1]
```diff
@@ -38,3 +38,106 @@ spec:
       service:
         # To Expose ArgoCD externally without ingress, set service type as "LoadBalancer". Default value is "ClusterIP".
         type: "LoadBalancer"
+  documentationURL: https://argoproj.github.io/cd/
```
Comment (xrstf) [inline pkg/ee/default-application-catalog/applicationdefinitions/argocd-app.yaml:41]: Why did you make all the YAML files executable?

2024-02-19T14-03-24

chmod 644 was good, why make it 755?

**Review by xrstf — CHANGES_REQUESTED** [review]
File system permissions

**Comment by archups** [comment]
> File system permissions

Fixed the file permissions. Thanks @xrstf

**Comment by SimonTheLeg** [comment]
/approve

With the filesystem fix done, looks good to me

**Review by embik — APPROVED** [review]

**Pull request merged**